zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -1251,6 +1251,25 @@ impl From<&circuit_input_builder::ExecStep> for ExecutionState {
                     OpcodeId::CALLDATALOAD => ExecutionState::CALLDATALOAD,
                     OpcodeId::CODESIZE => ExecutionState::CODESIZE,
                     OpcodeId::RETURN | OpcodeId::REVERT => ExecutionState::RETURN,
+                    // dummy ops
+                    OpcodeId::ADDRESS => ExecutionState::ADDRESS,
```

Comment:
I would ask to add a `warn` for each of those indicating the particular ExecutionState and mentioning that it's using a dummy gadget.
Like this:
```rust
                    OpcodeId::SHA3 => {
                        log::warn!("ExecutionState::SHA3 is implemented with DummyGadget");
                        ExecutionState::SHA3
                    }
```
And please do it as well for SHA3, which is currently using the dummy gadget.

I'm aware that there's a warning in the configure of the dummy gadget, but unfortunately this warning shows up even if the execution state doesn't appear in the trace used for the proof.  On the other hand, having a warning here will only appear if the trace uses that opcode; so I think that's preferable.